### PR TITLE
fix(background): don't clear overpaying on page refresh

### PR DESCRIPTION
**Pull request by sidvishnoi**
## Context

Closes https://github.com/interledger/web-monetization-extension/issues/450

### CODE REVIEW

File: src/background/services/tabEvents.ts [hunk 1]
```diff
@@ -75,11 +75,13 @@ export class TabEvents {
     tabId: TabId,
     changeInfo: Tabs.OnUpdatedChangeInfoType | Tabs.OnRemovedRemoveInfoType
   ) => {
+    // console.log('clearTabSessions', changeInfo)
     if (
       ('status' in changeInfo && changeInfo.status === 'loading') ||
       'isWindowClosing' in changeInfo
     ) {
-      this.monetizationService.clearTabSessions(tabId)
+      const clearOverpaying = !('isWindowClosing' in changeInfo) // TODO: verify this condition
```
Comment (sidvishnoi) [inline src/background/services/tabEvents.ts:83]: TODO

**Comment by raducristianpopa** [comment]
<h2>Extension builds preview</h2>

<table role="table">
<thead>
  <tr>
    <th>Name</th>
    <th>Link</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Latest commit</td>
    <td align="center">59bd9e52b173f95e439e7d7fbb97678316e1c9ca</td>
  </tr>
  <tr>
    <td align="center">Latest job logs</td>
    <td align="center"><a href="https://github.com/interledger/web-monetization-extension/actions/runs/10164377659">Run #10164377659</a></td>
  </tr>
  <tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/26595823857/artifacts/1755867726">Download</a></td></tr><tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/26595823857/artifacts/1755867729">Download</a></td></tr>
</tbody>
</table>

File: src/background/services/background.ts [hunk 2]
```diff
@@ -108,7 +108,9 @@ export class Background {
 
   bindTabHandlers() {
     this.browser.tabs.onRemoved.addListener(this.tabEvents.clearTabSessions)
-    this.browser.tabs.onUpdated.addListener(this.tabEvents.clearTabSessions)
+    this.browser.tabs.onUpdated.addListener(this.tabEvents.clearTabSessions, {
+      properties: ['url', 'discarded', 'status', 'isArticle']
+    })
```
Comment (raducristianpopa) [inline src/background/services/background.ts:113]: In Chrome:

image

File: src/background/services/tabEvents.ts [hunk 3]
```diff
@@ -75,12 +75,11 @@ export class TabEvents {
     tabId: TabId,
     changeInfo: Tabs.OnUpdatedChangeInfoType | Tabs.OnRemovedRemoveInfoType
   ) => {
-    // console.log('clearTabSessions', changeInfo)
     if (
       ('status' in changeInfo && changeInfo.status === 'loading') ||
       'isWindowClosing' in changeInfo
     ) {
-      const clearOverpaying = !('isWindowClosing' in changeInfo) // TODO: verify this condition
+      const clearOverpaying = 'isWindowClosing' in changeInfo
```
Comment (sidvishnoi) [inline src/background/services/tabEvents.ts:82]: Question: does it have to be `changeInfo.isWindowClosing === true`?

**Review by raducristianpopa — APPROVED** [review]

**Pull request merged**